api/cas-server-core-api-configuration-model/src/main/java/org/apereo/cas/configuration/model/core/CasServerProperties.java

```diff
@@ -39,6 +39,17 @@
     @RequiredProperty
     private String prefix = name.concat("/cas");
 
+    /**
+     * The name of the session cookie.
+     * 
+     * When using a distributed session store specially backed by CAS to primarily replicate CAS tokens and tickets across a cluster of CAS servers,
+     * the distribution mechanism needs to be made aware of session cookies. If a session is generated on a first node, when you reach the second node, 
+     * retrieving the session via the Tomcat session manager does not work. A {@code JSESSION_ID} cookie value is actually retrieved, but as there is no HTTP session
+     * associated, it is discarded and a new session with a new identifier is created. The only way to deal with that is to directly read the cookie value.
+     * The setting below allows one to customize the cookie name in such scenarios. 
+     */
+    private String sessionCookieName = "JSESSIONID";
```

Comment:
I think it might be best to move this setting into its own nested inner class here like the others; something like `SessionReplicationProperties`. This will make it clear that `sessionCookieName` is really only used for session replication. I imagine that the session-store will likely be enhanced with additional properties in the future, which would then be available in `SessionReplicationProperties`. (It's important that we try to optimize placement of properties as much as possible early on, as moving them around in the future generally is a not-so-pleasant op for upgrades)

PS I updated the description of the property based on your original notes in the PR.

Reply:
Sure, I can do that, but to be honest, I had the exact opposite thought by keeping the `sessionCookieName` outside the concept of replication, because I thought it could be re-used somewhere. It's unlikely though. I will do that.

Reply:
I have updated the PRs...